sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/rows.scala

```diff
@@ -201,7 +201,7 @@ class GenericRow(protected[sql] val values: Array[Any]) extends Row {
 
   override def toSeq: Seq[Any] = values.toSeq
 
-  override def copy(): Row = this
+  override def copy(): Row = new GenericRow(values.clone())
```

Comment:
Yes it is immutable, but if you want to extract the values of a GenericRows, change them and create a new row, then it's better to not change the original values. 
So you would first want to copy the original row. 

Also the method name copy() implies that it returns a real copy and not the reference on the original object.

Other rows like GenericInternalRow implement a correct the copy method:
  override def copy(): InternalRow = new GenericInternalRow(values.clone())